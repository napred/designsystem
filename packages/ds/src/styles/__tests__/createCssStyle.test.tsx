/**
 * @jest-environment jsdom
 */

import { css } from 'emotion';
import React from 'react';
// tslint:disable-next-line:no-implicit-dependencies
import { render } from 'react-testing-library';

import { createCssStyle } from '../';
import { Box, DesignSystem } from '../../';

describe('createCssStyle', () => {
  it('works with simple string', () => {
    const styles = [createCssStyle(['test'], 'color: blue;')];
    const { asFragment } = render(
      <DesignSystem styles={styles}>
        <Box />
      </DesignSystem>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('works with simple style', () => {
    const styles = [
      createCssStyle(
        ['test'],
        css`
          color: blue;
        `,
      ),
    ];
    const { asFragment } = render(
      <DesignSystem styles={styles}>
        <Box />
      </DesignSystem>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('works with simple object', () => {
    const styles = [createCssStyle(['test'], { color: 'blue' })];
    const { asFragment } = render(
      <DesignSystem styles={styles}>
        <Box />
      </DesignSystem>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('works with function', () => {
    const styles = [
      createCssStyle(
        ['test'],
        (props: { color: string }) =>
          css`
            color: ${props.color};
          `,
      ),
    ];
    const { asFragment } = render(
      <DesignSystem styles={styles}>
        <Box color="blue" />
      </DesignSystem>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
