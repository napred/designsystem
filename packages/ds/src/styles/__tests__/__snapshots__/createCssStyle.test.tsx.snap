// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createCssStyle works with function 1`] = `
<DocumentFragment>
  .css-ng7bw9 {
  color: #3B5998;
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  color: blue;
}

<div
    class="css-ng7bw9"
  />
</DocumentFragment>
`;

exports[`createCssStyle works with simple object 1`] = `
<DocumentFragment>
  .css-1f227de {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  color: blue;
}

<div
    class="css-1f227de"
  />
</DocumentFragment>
`;

exports[`createCssStyle works with simple string 1`] = `
<DocumentFragment>
  .css-14q17fo {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  color: blue;
}

<div
    class="css-14q17fo"
  />
</DocumentFragment>
`;

exports[`createCssStyle works with simple style 1`] = `
<DocumentFragment>
  .css-8xq7br {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  color: blue;
}

<div
    class="css-8xq7br"
  />
</DocumentFragment>
`;
